vault/resource_gcp_secret_impersonated_account_test.go

```diff
@@ -0,0 +1,191 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+	"golang.org/x/oauth2/google"
+)
+
+// This test requires that you pass credentials for a user or service account having the IAM rights
+// listed at https://www.vaultproject.io/docs/secrets/gcp/index.html for the project you are testing
+// on. The credentials must also allow setting IAM permissions on the project being tested.
+func TestGCPSecretImpersonatedAccount(t *testing.T) {
+	backend := "gcp"
+	impersonatedAccount := acctest.RandomWithPrefix("tf-test")
+	credentials, project := testutil.GetTestGCPCreds(t)
+
+	// We will use the provided key as the impersonated account
+	conf, err := google.JWTConfigFromJSON([]byte(credentials), "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		t.Fatalf("error decoding GCP Credentials: %v", err)
+	}
+	serviceAccountEmail := conf.Email
+
+	noBindings := testGCPSecretImpersonatedAccount_accessToken(backend, impersonatedAccount, credentials, serviceAccountEmail)
+
+	resourceName := "vault_gcp_secret_impersonated_account.test"
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testGCPSecretImpersonatedAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: noBindings,
+				Check: resource.ComposeTestCheckFunc(
+					testGCPSecretImpersonatedAccount_attrs(backend, impersonatedAccount),
```

Comment:
This is an older style of writing tests in the codebase. For new resource additions, we prefer to have a test structure  that is something [along the lines of this](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kv_secret_v2_test.go#L24-L64)

As shown in the test above, the complete loop of testing contains an initial test step, an update test step, and finally an import test step. The first two test cases confirm that values are obtained from the TF config as expected (`.tf` file). The Import test step verifies that the expected values were written to Vault, and the values that are read from Vault are what we had initially obtained from the TF config. 

In this case, the checks you have from L46-L51 are accurate, but we can remove L45 entirely, since this check adds a lot of repeated code to the codebase and we are trying to move away from this style for new resources.
```suggestion
```